src/library.tsx

```diff
@@ -198,7 +203,10 @@ const RenderWidgetShell = (props: IInit) => {
     <div className={`fixed ${classes.containerClassName}`}>
       <div
         className={`${classes.widgetContainerClassName} rounded-full bg-black flex items-center justify-center cursor-pointer`}
-        onClick={() => setIsOpen(!isOpen)}
+        onClick={() => {
+          setIsOpen(!isOpen);
+          setTerminalInView(!getTerminalInView());
+        }}
```

Comment:
if `isOpen` and `getTerminalInView()` is not in sync, you will end up in crazy state toggling that never matches up
you should rely on `isOpen` state as the source of truth, so it stays in sync

```
() => {
  if (isOpen) {
    setIsOpen(false);
    setTerminalInView(false);
  } else {
     // ... true true
  }
 
}
```